src/components/Wallet.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import AddSigners from './AddSigners';
+import Signers from './Signers';
+import { Device } from '../types'
+import { AppState } from '../store';
+import { getWallets, selectCandidateDevicesForActiveWallet, selectActiveWallet } from '../store/wallet';
+import { Spinner } from 'reactstrap';
+import api from '../api'
+
+interface StateProps {
+  candidateDevices: Device[];
+  deviceError: Error | null;
+  activeWallet: AppState['wallet']['activeWallet'];
+}
+
+interface DispatchProps {
+  getWallets: typeof getWallets;
+}
+
+interface State {
+  deviceBeingAdded: Device | null;
+}
+
+type Props = StateProps & DispatchProps;
+
+class Wallet extends React.Component<Props, State> {
+  state: State = {
+    deviceBeingAdded: null,
+  }
+  private addSigner = async (device: Device) => {
+    if (this.props.activeWallet) {
+      this.setState({ deviceBeingAdded: device })
+      await api.addSigner({
+        wallet_name: this.props.activeWallet.name,
+        signer_name: device.type,
+        device_id: device.fingerprint,
+      })
+      // FIXME: state.wallet.activeWallet goes stale here
+      await this.props.getWallets();
```

Comment:
Action functions aren't async, so no need to await on them. There's no harm in adding an await though, just unnecessary.